### PR TITLE
Fix partition id in the fp32->fp16 param copying step for z2+cpu-offload

### PR DESCRIPTION
**Pull request by siddharth9820**
This is the same fix as this [PR](https://github.com/microsoft/DeepSpeed/pull/2058) for cpu-offloading. Here's the corresponding PR in the Megatron-Deepspeed repo that adds cpu-offloading support -  [link](https://github.com/microsoft/Megatron-DeepSpeed/pull/55) 

Here is the loss curve after the fix compared with zero-stage 0 for the following setting - 

Base Model - 1.3B
Number of Experts - 8
Batch Size - 256
Machine - Azure A100 40GB
Number of GPUs - 8
Dataset - BookCorpus

image

### CODE REVIEW

File: deepspeed/runtime/zero/stage_1_and_2.py [hunk 1]
```diff
@@ -1746,13 +1746,19 @@ def step(self, closure=None):
...
+                        bit16_partitions[dist.get_rank(
+                            group=self.real_dp_process_group[group_id])]
+                    ] for group_id,
+                    bit16_partitions in enumerate(self.parallel_partitioned_bit16_groups)
+                ]
                 self.optimizer.step(fp16_param_groups=bit16_param_groups)
             else:
                 self.optimizer.step()
                 for bit16_partitions, fp32_partition in zip(self.parallel_partitioned_bit16_groups, self.single_partition_of_fp32_groups):
+                    partition_id = dist.get_rank(
```
Comment (tjruwase) [inline deepspeed/runtime/zero/stage_1_and_2.py:1760]: Can you define `partition_id` and resue for both cases?

Comment (siddharth9820) [reply deepspeed/runtime/zero/stage_1_and_2.py:1760]: Done
